sgx-types/src/lib.rs

```diff
@@ -73,28 +73,12 @@ pub mod tcs;
 
 use core::marker::PhantomData;
 
-/// An offset reference with neither read nor write capabilities
-///
-/// The Offset struct allows the creation of an opaque reference to a
-/// type that cannot be read or written. Neither the lifetime nor the
-/// type are discarded. This allows us to refer to an offset inside
-/// an enclave without fear that it will be dereferenced. The size of
-/// the Offset is always 64 bits with natural alignment. Therefore,
-/// the Offset type can be embedded in structs.
 #[repr(transparent)]
 #[derive(Debug)]
-pub struct Offset<'h, T>(u64, PhantomData<&'h T>);
+pub struct Offset<T: ?Sized>(u64, PhantomData<T>);
```

Comment:
Looks like this removes the lifetime and also the `Sized` bound, which I don't totally understand yet - why do we not want `T` to be `Sized`?

Reply:
Because `[u8]` is not Sized. This is helpful for the `add_slice()` function.